File: src/stringutil.py
import re

RE_SPLIT_WORDS = re.compile("\s+|[,.!?:]+(\s|$)+", re.M)
RE_SPLIT_SENTENCES = re.compile("(?<=[.!?])(\s+)", re.M)


def split_words(s):
    return remove_empty(RE_SPLIT_WORDS.split(s))


def split_sentences(s):
    return remove_empty(RE_SPLIT_SENTENCES.split(s))


def remove_empty(str_arr):
    """
    >>> remove_empty(['', ' ', '\\n', None, 'hello'])
    ['hello']
    """
    return list(
        filter(lambda x: x is not None and not x.isspace() and x is not '', str_arr))

File: src/stringutil.js
"use strict";

const RE_SPLIT_WORDS = /\s+|[,.!?:]+(\s|$)+/m;
const RE_SPLIT_SENTENCES = /(?<=[.!?])(\s+)/m;

/**
 * Extracts words from the given string.
 *
 * @param {string} str String to split
 * @returns {Array.<String>} Words from the string
 */
function splitWords(str) {
  return removeEmpty(str.split(RE_SPLIT_WORDS));
}

/**
 * Extracts sentences from the string
 *
 * @param {string} str String to split
 * @returns {Array.<String>} Sentences from the string
 */
function splitSentences(str) {
  return removeEmpty(str.split(RE_SPLIT_SENTENCES));
}

module.exports = {
  splitWords: splitWords,
  splitSentences: splitSentences
};

/**
 * Removes undefined, null and empty strings from the given array.
 *
 * @param {Array.<String>} strArr The array
 * @returns {Array.<String>} Array without null or undefined elements, or empty strings
 */
function removeEmpty(strArr) {
  return strArr.filter(el => el !== null && el !== undefined && el.trim() !== "");
}